lumicks/pylake/calibration.py

```diff
@@ -1,3 +1,224 @@
+import re
+from collections import UserDict
+
+
+def _read_from_bl_dict(bl_dict):
+    """Returns a dictionary of calibration parameters from a calibration dictionary exported from
+    Bluelake where the keys respond to parameter names used in Pylake.
+
+    Parameters
+    ----------
+    bl_dict : dict[str, float]
+        Raw dictionary coming from a calibration item.
+    """
+
+    mapping = {
+        "bead_diameter": "Bead diameter (um)",
+        "rho_bead": "Bead density (Kg/m3)",
+        "rho_sample": "Fluid density (Kg/m3)",
+        "viscosity": "Viscosity (Pa*s)",
+        "temperature": "Temperature (C)",
+        "driving_frequency_guess": "Driving data frequency (Hz)",
+        "distance_to_surface": "Bead center height (um)",
+        # Fixed diode parameters (only available when diode was fixed)
+        "fixed_alpha": "Diode alpha",
+        "fixed_diode": "Diode frequency (Hz)",
+        # Only available for axial with active
+        "drag": "gamma_ex_lateral (kg/s)",
+    }
+
+    return {
+        key_param: bl_dict[key_bl] for key_param, key_bl in mapping.items() if key_bl in bl_dict
+    }
+
+
+def _verify_full(method):
+    def wrapper(self, *args, **kwargs):
+        if (kind := self.data.get("Kind", "Unknown")) != "Full calibration":
+            raise ValueError(
+                "These parameters are only available for a full calibration. Instead, this is "
+                f"a calibration item where the following operation was applied: {kind}."
+            )
+
+        return method(self, *args, **kwargs)
+
+    return wrapper
```

Comment:
you can even keep this scoped locally in the class, which I think might be better because it relies on that specific class (for now at least)

Reply:
Sounds good. Should also still include `wraps` to forward the docstring.